src/NaturalEarth.jl

```diff
@@ -50,5 +65,195 @@ function bathymetry(contour::Int=2000)
+#     retval = Dict{String, Vector{VersionNumber}}()
+#     for key_element in keys
+#         key_str = key_element[1]
+#         endswith(key_str, "/") && continue # discard directory specs
+#         if haskey(key_str, "raster")
+#             name = splitext(basename(key_str))[1]
+#             dirs = splitpath(name)
+
+#             version = if length(dirs) == 1 || !(occursin(".", dirs[1])) # no version number
+#                 VersionNumber(typemax(UInt32))
+#             else
+#                 VersionNumber(dirs[1])
+#             end
+
+#             # for i in 1
+#     end
+# end
+
+# Implement the RasterDataSources.jl interface.
+# With this, you can call e.g. # Rasters.Raster(NaturalEarthRaster{:latest, 10, :HYP_HR}, nothing)
+
+struct NaturalEarthRaster{Version, Scale, Name} <: RasterDataSources.RasterDataSource end
+
+function ne_raster(; version = :latest, scale::Int = 10, name::Int) 
+    NaturalEarthRaster{Symbol(version), scale, Symbol(name)}
+end
+
+const _NATURALEARTH_URI = RasterDataSources.URIs.URI(scheme="https", host="naturalearth.s3.amazonaws.com", path="/")
+
+function _zippath(::Type{NaturalEarthRaster{Version, Scale, Name}}) where {Version, Scale, Name}
+    return if Version == :latest || VersionNumber(string(Version)).major == typemax(UInt32)
+        "" # max version, so latest
+    else
+        string(Version) * "/"
+    end * if Scale == 110 || Scale == 50 || Scale == 10
+        string(Scale) * "m_raster"
+    else
+        error("Invalid scale: $Scale")
+    end * "/" * string(Name) * ".zip"
+end
+
+_zipfile_to_read(raster_name, zf) = first(filter(f -> splitdir(f.name)[2] == raster_name, zf.files))
+
+function RasterDataSources.zippath(::Type{NaturalEarthRaster{Version, Scale, Name}}) where {Version, Scale, Name}
+    return joinpath(RasterDataSources.rasterpath(), "NaturalEarth", "zips", split(_zippath(NaturalEarthRaster{Version, Scale, Name}), "/")...)
+end
+
+function RasterDataSources.zipurl(::Type{NaturalEarthRaster{Version, Scale, Name}}) where {Version, Scale, Name}
+    return RasterDataSources.URIs.URI(_NATURALEARTH_URI; path = "/" * _zippath(NaturalEarthRaster{Version, Scale, Name}))
+end
+
+function RasterDataSources.zipname(::Type{NaturalEarthRaster{Version, Scale, Name}}) where {Version, Scale, Name}
+    return splitpath(_zippath(NaturalEarthRaster{Version, Scale, Name}))[end]
+end
+
+function RasterDataSources.rasterpath(T::Type{NaturalEarthRaster{Version, Scale, Name}}) where {Version, Scale, Name}
+    return joinpath(RasterDataSources.rasterpath(), "NaturalEarth", split((_zippath(NaturalEarthRaster{Version, Scale, Name})), "/")[1:end-1]..., RasterDataSources.rastername(T))
+end
+
+function RasterDataSources.rastername(::Type{NaturalEarthRaster{Version, Scale, Name}}) where {Version, Scale, Name}
+    return string(Name) * ".tif"
+end
+
+function RasterDataSources.getraster(T::Type{NaturalEarthRaster{Version, Scale, Name}}, layer = nothing) where {Version, Scale, Name}
```

Comment:
I would just have a `scale` keyword so there is no recompilation.

Reply:
```julia
Raster(NaturalEarth, :HY_SR; scale=10)
```

Reply:
Ok so turns out that wont work! Because Rasters.jl has a fixed list of keywords to pass to `getraster` which does not include `scale`.

That's not going to "scale". I think we need a `RasterDataSource` to declare its keywords to make this more flexible.

Like:
```julia
getraster_keywords(NaturalEarth) = (:scale, :version) 
```

Then Rasters.jl can just pass those through and keep the rest.

Reply:
Yeah, I'd also like to discuss the `getraster` interface more generally...the reliance on types kind of confused me

Reply:
With no types at we have no dispatch. But do you mean types rather than objects?

Partly its just historical, I didnt start the package.

Reply:
https://github.com/EcoJulia/RasterDataSources.jl/pull/57